### ISSUE TITLE
Build error, generic inheritance issue

### ISSUE THREAD

**Issue by tomasbrod** [GregTechCE/GregTech#25]
```
> Task :compileJava 
.../GregTechCE/build/sources/main/java/gregtech/api/util/GTControlledRegistry.java:11: error: Iterable cannot be inherited with different arguments: <> and <T>
public class GTControlledRegistry<T> extends RegistryNamespaced<String, T> {
       ^
```
With system-wide Gradle 4.6 and java 1.8.0_162 (java 8 openjdk).

**Comment by Exidex** [comment]
Did you run setupDecompWorkspace?

**Comment by temp1011** [comment]
I get some errors like:
`ERROR Unknown Type: MetaItem<?>.MetaValueItemclass org.eclipse.jdt.core.dom.QualifiedType 510|51025
`
That might be due to IDEA though

**Comment by Exidex** [comment]
Yeah, errors like
 ```
ERROR Unknown Type: MetaItem<?>.MetaValueItemclass org.eclipse.jdt.core.dom.QualifiedType 9866|986625
ERROR Unknown Type: ToolMetaItem<?>.MetaToolValueItemclass org.eclipse.jdt.core.dom.QualifiedType 18129|1812933
```
appear when building jar even though build completes ok. I wasn't able to figure out what is it or how to fix it. 

The compile error with `GTControlledRegistry` seems to appear when you run `setupCiWorkspace` or `setupDevWorkspace` (or when you try to build jar with out setting up workspace first) and I also wasn't able to figure out why. `setupDecompWorkspace` works fine though. Because of this compile error I had to setup travis with `setupDecompWorkspace` and with it travis build runs for 7-9 minutes which is a lot and should be fixed. 
Any help is appreciated.

**Comment by tomasbrod** [comment]
After deleting artifacts and cache, numerous combinations of gradle and setupXXXWorkspace, it finally compiled. I guess it needs some articfacts from previous invocation of one one of the commands which are not preoperly set to depend on.

**Comment by Exidex** [comment]
with travis every time is a first time because caches are removed after each build

**Comment by OvermindDL1** [comment]
> with travis every time is a first time because caches are removed after each build

Not unless you have a `cache` key setup in the .travis.yml file (which can save substantial time when well used).

**Comment by Exidex** [comment]
the real question is why it doesn't work with the same configuration as in other mods.

**Comment by mightydanp** [comment]
Can someone  give out a compiled version ive been trying aswell to compile it so i could look at the progress thos far

**Comment by Exidex** [comment]
I've fixed that compile error in 05ea5ec101715fced7510f21b9bb799429d138d6. The `ERROR Unknown Type` error doesn't cause any problems so I am gonna leave it as is for now. 
@mightydanp you should be able to compile it yourself or wait for alpha builds
